app/models/petition.rb

```diff
@@ -156,6 +155,13 @@ def self.counts_by_state
     counts_by_state
   end
 
+  def publish!
+    self.state = Petition::OPEN_STATE
+    self.open_at = Time.zone.now
+    self.closed_at = AppConfig.petition_duration.months.from_now.end_of_day
```

Comment:
Could change it to ``(AppConfig.petition_duration.months.from_now + 1.day).midnight`` which would be 1 second after ``AppConfig.petition_duration.months.from_now.end_of_day`` with the advantage of 0-ing all the hours, minutes, seconds, and fractions.